### PR TITLE
Update PR Build Action for API

### PR DESCRIPTION
**Pull request by Kas-tle**
Because the API is now in a separate repo, PRs that rely on API changes will fail to build with the current PR build action. This PR updates the action to check if the user initiating the PR has a fork of the API repo with a branch matching the name of their PR. If this is the case, the action will clone the users API repo and cache it to the action runner's local maven repo before building Geyser. As a result, the action can then use the user's changes to the API when building Geyser.

Validation: https://github.com/Kas-tle/Geyser/pull/4

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -48,4 +48,6 @@ Make sure to comment your code where possible.
 
 The nature of our software requires a lot of arrays and maps to be stored - where possible, use Fastutil's specialized maps. For example, if you're storing block state translations, use an `Int2IntMap`.
 
-We have a rundown of all the tools you need to develop over on our [wiki](https://github.com/GeyserMC/Geyser/wiki/Developer-Guide). If you have any questions, please feel free to reach out to our [Discord](https://discord.gg/geysermc)!
+We have a rundown of all the tools you need to develop over on our [wiki](https://wiki.geysermc.org/other/developer-guide/). If you have any questions, please feel free to reach out to our [Discord](https://discord.gg/geysermc)!
+
+If making a pull request that also depends on changes to [Geyser's API](https://github.com/GeyserMC/api), simply fork the API repo and create a branch with the same name as your PR to Geyser. The pull request [action](https://github.com/GeyserMC/Geyser/blob/master/.github/workflows/pullrequest.yml) will automatically cache your Geyser API changes before building your changes to Geyser.
```
Comment (Tim203) [inline CONTRIBUTING.md:53]:
```suggestion
If you're making a pull request that also depends on changes to [the base API](https://github.com/GeyserMC/api), simply fork the API repo and create a branch with the same name as your Geyser PR. The pull request [action](https://github.com/GeyserMC/Geyser/blob/master/.github/workflows/pullrequest.yml) will automatically use your API changes while building your changes to Geyser.
```

File: .github/workflows/pullrequest.yml [hunk 2]
```diff
@@ -15,6 +15,22 @@ jobs:
           distribution: 'temurin'
           java-version: 17
           cache: 'gradle'
+      - name: Check if PR user has API repo
```
Comment (Tim203) [inline .github/workflows/pullrequest.yml:18]:
```suggestion
      - name: Check if the user has forked the API repo
```
or `author` instead of `user`

**Review by Tim203 — CHANGES_REQUESTED** [review]
Nitpicks that might clarify the process

File: .github/workflows/pullrequest.yml (hunk 2 above)
Comment (Kas-tle) [reply .github/workflows/pullrequest.yml]: https://github.com/GeyserMC/Geyser/pull/3535/commits/5ebfc0f361518d72b5c69c9f0b46899ff9b3e9ae#diff-ce9a2f9036cc2a5aaf1f78839c4f65c7a12808822e0dd0f3552471921ac5e007L18-R18

File: .github/workflows/pullrequest.yml [hunk 3]
```diff
@@ -15,6 +15,22 @@ jobs:
...
+        uses: Kas-tle/ForkFinder@v1.0.1
+        id: find_forks
+        with:
+          owner: 'GeyserMC'
+          repo: 'api'
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Cache PR user API repo
```
[the remark this answers is not in the document]
Comment (Kas-tle) [reply .github/workflows/pullrequest.yml]: https://github.com/GeyserMC/Geyser/pull/3535/commits/5ebfc0f361518d72b5c69c9f0b46899ff9b3e9ae#diff-ce9a2f9036cc2a5aaf1f78839c4f65c7a12808822e0dd0f3552471921ac5e007L25-R25

File: CONTRIBUTING.md (hunk 1 above)
Comment (Kas-tle) [reply CONTRIBUTING.md]: https://github.com/GeyserMC/Geyser/pull/3535/commits/5ebfc0f361518d72b5c69c9f0b46899ff9b3e9ae#diff-ce9a2f9036cc2a5aaf1f78839c4f65c7a12808822e0dd0f3552471921ac5e007L25-R25

**Comment by Kas-tle** [comment]
Already merged manually by Tim so closing.

**Comment by Kas-tle** [comment]
Wait never mind...

**Review by Tim203 — APPROVED** [review]

**Pull request merged**